### PR TITLE
Support generating sphinx doc

### LINKED ISSUE #162: Setup readthedocs
**What's the problem this feature will solve?**

Website for presentation documentation

**Describe the solution you'd like**

https://cell-locator.readthedocs.io

Adapt what was done in Slicer repo

**Alternative Solutions**

NA

**Additional context**

NA

### PR DESCRIPTION
**Pull request by allemangD**
Notable changes I've made since @jcfr's contributions:

- Migrated content from the Wiki and some PRs to the documentation (see fa60c31)
- Cleared that documentation from the github wiki; so the wiki could be disabled once these changes are published
- Switched to MyST markdown parser, mentioned in https://github.com/BICCN/cell-locator/issues/162#issuecomment-852458054, and converted all existing docs to markdown.

One note regarding MyST - if we ever want to do reference-style documentation of the classes (see [pillow](https://pillow.readthedocs.io/en/stable/reference/Image.html)'s api reference, for example), we will still have to use reStructuredText. These types of docs use one or both of [sphinx.ext.autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) and [sphinx python domain](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#the-python-domain), neither of which is supported by MyST. I began creating such documentation to replace [this code block](https://github.com/BICCN/cell-locator/blob/798e7ae670bc2c79472949ed2863b3d57a8926f5/Documentation/developer_guide/Annotations.md#creating-annotation-types) and ran into the issue.

Docs are published at https://cell-locator.readthedocs.io/en/support-generating-sphinx-doc/

Resolves #162.

### CODE REVIEW

**Comment by allemangD** [comment]
Also worth noting there is a [sphinx c++ domain](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cpp-domain) for reference documentation.

**Comment by allemangD** [comment]
There is a workaround to render reference docs and autodoc, mentioned here; enclose the problematic section in an `eval-rst` directive.

https://github.com/executablebooks/MyST-Parser/issues/163#issuecomment-679907892

**Comment by jcfr** [comment]
> and I don't think there is any autodoc available like for Python

Indeed. That said, there is the possibility to integrate with doxygen using https://breathe.readthedocs.io/en/latest/

File: README.md [hunk 1]
```diff
@@ -1,163 +1,11 @@
...
+# CellLocator by Allen Institute
 
-Manually aligning specimens to annotated 3D spaces.
+Cell Locator is a free, open source and multi-platform pinning tool to facilitate
+mapping samples into common 3D spaces.
 
-Desktop  application based on 3D Slicer that displays the AIBS CCF or MNI atlas, color segmented
-by structure of the brain, and allows a user to create a planar polyline annotation to
-facilitate mapping samples into the corresponding atlas.
+It is a desktop application based on 3D Slicer that displays the Allen Institue for Brain Science
```
Comment (jcfr) [inline README.md:6]:
```suggestion
It is a desktop application based on 3D Slicer that displays the Allen Institute for Brain Science
```

File: Documentation/developer_guide/LIMSIntegration.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# LIMS Integration
+
+Integration with a Laboratory Information Management System (LIMS) is acheived through a simple REST api; there are only two endpoints to get and set annotations for a particular specimen.
```
Comment (jcfr) [inline Documentation/developer_guide/LIMSIntegration.md:3]:
```suggestion
Integration with a Laboratory Information Management System (LIMS) is achieved through a simple REST api; there are only two endpoints to get and set annotations for a particular specimen.
```

**Review by jcfr — APPROVED** [review]

**Comment by jcfr** [comment]
I updated readthedocs setting and the currently active version is now "latest"
see https://cell-locator.readthedocs.io/en/latest/

**Pull request merged**